src/naive/internals.rs

```diff
@@ -15,14 +15,17 @@
 
 #![cfg_attr(feature = "__internal_bench", allow(missing_docs))]
 
-use crate::Weekday;
+use crate::{expect, Weekday};
 use core::fmt;
+use core::num::NonZeroI32;
 
-/// The internal date representation: `year << 13 | Of`
-pub(super) type DateImpl = i32;
+/// The internal date representation: `year << 13 | Of`.
+/// Because the ordinal in `Of` is always in the range 1..=366, `DateImpl` will never be `0`.
+/// The year flags can also never be `0`.
+pub(super) type DateImpl = NonZeroI32;
```

Comment:
I'm open to this change, but I'd like another commit before this that changes `DateImpl` to be a newtype `struct` rather than a type alias. Ideally we can then add some methods (like `new()`) that avoids a bunch of this repetition.

Reply:
Much better, I should have gone with this in the first place :+1:.